Modules/_winapi.c

```diff
@@ -502,7 +502,7 @@ _winapi_CreateFileMapping_impl(PyObject *module, HANDLE file_handle,
     Py_END_ALLOW_THREADS
 
     if (handle == NULL) {
-        PyErr_SetFromWindowsErrWithUnicodeFilename(0, name);
+        PyErr_SetFromWindowsErr(0);
```

Comment:
This change removes the name information which is useful, no?

Can PyErr_SetExcFromWindowsErrWithFilenameObjects() be used with PyUnicode_FromWideChar() here? Basically, copy-paste PyErr_SetFromWindowsErrWithUnicodeFilename() code:

```
    PyObject *name = filename ? PyUnicode_FromWideChar(filename, -1) : NULL;
    PyObject *result = PyErr_SetExcFromWindowsErrWithFilenameObjects(
                                                  PyExc_OSError,
                                                  ierr, name, NULL);
    Py_XDECREF(name);
    return result;
```

These function names are super long. I dislike these APIs :-(

Reply:
> Can PyErr_SetExcFromWindowsErrWithFilenameObjects() be used with PyUnicode_FromWideChar() here?

Yes. Eryk Sun had suggested the same thing on the bug tracker.